test/solc/CommandLineParser.cpp

```diff
@@ -422,6 +420,32 @@ BOOST_AUTO_TEST_CASE(standard_json_mode_options)
 	BOOST_TEST(parsedOptions.value() == expectedOptions);
 }
 
+BOOST_AUTO_TEST_CASE(experimental_via_ir_invalid_input_modes)
+{
+	vector<string> commandLine = {
+		"solc",
+		"--experimental-via-ir",
+		"file",
+	};
+
+	static vector<vector<string>> const allowedCombinations = {
+		{"--assemble"},
+		{"--yul"},
+		{"--strict-assembly"},
+		{"--standard-json"},
+		{"--link"}
+	};
+	for (auto const& assemblyOptions: allowedCombinations)
+	{
+		stringstream sout, serr;
+		optional<CommandLineOptions> parsedOptions = parseCommandLine(commandLine + assemblyOptions, sout, serr);
```

Comment:
```suggestion
		optional<CommandLineOptions> parsedOptions = parseCommandLine(commandLine + {inputModeOption}, sout, serr);
```

If this does not work, you can also just move `commandLine` variable into the loop.